src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -317,7 +326,17 @@ impl Coordinator {
                     self.optimizer_metrics(),
                 ))
             }
-            Some(copy_to_ctx) => {
+            Some(mut copy_to_ctx) => {
+                // Getting the max worker count across replicas
+                // and using that value for the number of batches to
+                // divide the copy output into.
+                let max_worker_count = cluster
+                    .replicas()
+                    .map(|r| r.config.location.workers())
+                    .max()
+                    .map(u64::cast_from);
+                assert!(max_worker_count.is_some());
```

Comment:
I'd feel better if this computation was incorporated in the check above such that they never get out of sync
```suggestion
        let max_worker count = match cluster.replicas().map(|r| r.config.location.workers()).max() {
            Some(count) => u64::cast_from(count),
            None => {
                return Err(AdapterError::NoClusterReplicasAvailable(
                    cluster.name.clone(),
                ));
            }
        }
```